apps/dashboard/src/components/workflow-editor/steps/push/push-preview.tsx

```diff
@@ -0,0 +1,126 @@
+import { HTMLAttributes } from 'react';
+import { HTMLMotionProps, motion } from 'motion/react';
+import { ChannelTypeEnum, GeneratePreviewResponseDto } from '@novu/shared';
+import { Skeleton } from '@/components/primitives/skeleton';
+import { cn } from '@/utils/ui';
+
+export function PushPreview({
+  isPreviewPending,
+  previewData,
+}: {
+  isPreviewPending: boolean;
+  previewData?: GeneratePreviewResponseDto;
+}) {
+  if (isPreviewPending) {
+    return (
+      <PushBackgroundWithPhone>
+        <PushNotificationContainer>
+          <PushContentContainerPreview className="relative z-10">
+            <PushSubjectPreview isPending />
+            <PushBodyPreview isPending />
+          </PushContentContainerPreview>
+        </PushNotificationContainer>
+      </PushBackgroundWithPhone>
+    );
+  }
+
+  if (previewData?.result.type !== ChannelTypeEnum.PUSH) {
```

Comment:
Don't we already have guarantees in parent components that this push-preview will only be rendered when the channel type is a PUSH?

If yes, we can remove this conditional.

Reply:
Yes, but this also helps TS narrowing the types

Reply:
hahaha Ok, not a good enough reason but not a blocker as well. If we change the preview DTO to a much flatter alternative it would be much easier for us to fix this issue.